src/common/FrameManager.ts

```diff
@@ -218,18 +225,53 @@ export class FrameManager extends EventEmitter {
     return watcher.navigationResponse();
   }
 
-  private async _onFrameMoved(event: Protocol.Target.AttachedToTargetEvent) {
+  private async _onAttachedToTarget(
+    event: Protocol.Target.AttachedToTargetEvent
+  ) {
     if (event.targetInfo.type !== 'iframe') {
       return;
     }
 
-    // TODO(sadym): Remove debug message once proper OOPIF support is
-    // implemented: https://github.com/puppeteer/puppeteer/issues/2548
-    debug('puppeteer:frame')(
-      `The frame '${event.targetInfo.targetId}' moved to another session. ` +
-        `Out-of-process iframes (OOPIF) are not supported by Puppeteer yet. ` +
-        `https://github.com/puppeteer/puppeteer/issues/2548`
+    const frame = this._frames.get(event.targetInfo.targetId);
+    const session = Connection.fromSession(this._client).session(
+      event.sessionId
     );
+    frame.updateClient(session);
+    session.on('Page.frameNavigated', (event) => {
```

Comment:
should not we clean up listeners on detach?

Reply:
The session will be destroyed anyways if the frame disappears but I'll see if can include clean up to make it more future proof.

Reply:
I'll keep it as it as it follows the current approach but I'll investigate further in the future.